packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -185,6 +186,31 @@ export class WorkspaceCommandContribution implements CommandContribution {
     @inject(WorkspaceDuplicateHandler) protected readonly duplicateHandler: WorkspaceDuplicateHandler;
     @inject(WorkspaceCompareHandler) protected readonly compareHandler: WorkspaceCompareHandler;
 
+    private readonly onDidCreateNewFileEmitter = new Emitter<URI>();
+    private readonly onDidCreateNewFolderEmitter = new Emitter<URI>();
+
+    /**
+     * Emit when the 'New File' command is clicked.
+     */
+    get onNewFileCommand(): Event<URI> {
+        return this.onDidCreateNewFileEmitter.event;
+    }
+
+    /**
+     * Emit when the 'New Folder' command is clicked.
```

Comment:
i think this comment can be removed, commands are generic APIs and can be called programmatically, they don't necessary require user interactions, the same for another event